libs/translator/reader/utils.cpp

```diff
@@ -697,21 +698,21 @@ void ReadCameraShaders(const UsdPrim& prim, AtNode *node, UsdArnoldReaderContext
 int ComputeTransformNumKeys(const UsdPrim &prim, const TimeSettings &time, bool checkParents) {
     // No motion blur, need just 1 key
     if (!time.motionBlur) return 1;
-
-    int numKeys = 2; // We need at least 2 keys at the interval boundaries
-
-    // The deform keys attribute takes precedence on any prims
-    if (UsdAttribute deformKeysAttr = prim.GetAttribute(_tokens->PrimvarsArnoldDeformKeys)) {
-        UsdGeomPrimvar primvar(deformKeysAttr);
-        if (primvar) {
-            int deformKeys = 0;
-            if (deformKeysAttr.Get(&deformKeys, UsdTimeCode(time.frame))) {
-                numKeys = deformKeys > 0 ? deformKeys : 1;
-            }
-            return numKeys;
+    const auto getNumKeys = [&](const UsdPrim &prim, const TfToken& attr, float frame) -> int {
```

Comment:
prim argument is not needed it's already passed as reference, and time.frame as well.